File: src/library/java/gg/generations/rarecandy/renderer/rendering/StateToggle.java
package gg.generations.rarecandy.renderer.rendering;

public class StateToggle {
    private final Runnable enable;
    private final Runnable disable;
    private boolean state;
    public StateToggle(Runnable enable, Runnable disable) {
        this.enable = enable;
        this.disable = disable;
    }

    public void toggle(boolean newState) {
        if(newState && !state) {
            state = true;
            enable.run();
        } else if(!newState && state) {
            state = false;
            disable.run();
        }

    }
}

File: src/library/java/gg/generations/rarecandy/renderer/rendering/StateToggle.kt
package gg.generations.rarecandy.renderer.rendering

class StateToggle(private val enable: () -> Unit, private val disable: () -> Unit, private var state : Boolean = false) {

    fun toggle(newState: Boolean) {
        if (newState && !state) {
            state = true
            enable.invoke()
        } else if (!newState && state) {
            state = false
            disable.invoke()
        }
    }
}
